### PR TITLE
Exams Yulianna Vakhnina IM-32

### PR DESCRIPTION
**Pull request by yuliakinda**
Tasks done:

between.js
count.js
day.js
drop.js
duplicate.js
except.js
inc.js
intersection.js
invert.js
ip.js

### CODE REVIEW

File: Tasks/between.js [hunk 1]
```diff
@@ -1,21 +1,26 @@
...
-      i = str.indexOf(s);
-      if (i === -1) {
-        return '';
-      } else {
-        str = str.substring(0, i);
-      }
-    }
+  
+  let k = i + prefix.length;
+  inputString = inputString.substring(k);
```
Comment (tshemsedinov) [inline Tasks/between.js:13]: Do not change incoming args

File: Tasks/between.js [hunk 2]
```diff
@@ -1,21 +1,26 @@
...
+  
+  if (suffix) {
+    const i = inputString.indexOf(suffix);
+    if (i === -1) return '';
+    inputString = inputString.substring(0, i);
```
Comment (tshemsedinov) [inline Tasks/between.js:18]: Ditto

File: Tasks/between.js [hunk 3]
```diff
@@ -1,21 +1,26 @@
...
   }
-  return str;
+
+  const substring = inputString;
+
+  return substring;
```
Comment (tshemsedinov) [inline Tasks/between.js:23]: ???

File: Tasks/drop.js [hunk 4]
```diff
@@ -1,24 +1,20 @@
...
+// remove unneeded parts, replace forEach with for...of
+
+'use strict';
+
+const drop = (dictionary, ...keysToRemove) => {
+  const keys = Object.keys(dictionary);
+
+  for (const key of keys) {
+    if (keysToRemove.includes(key)) {
+      delete dictionary[key];
```
Comment (tshemsedinov) [inline Tasks/drop.js:13]: Dry to implement without delete

File: Tasks/duplicate.js [hunk 5]
```diff
@@ -1,14 +1,13 @@
...
+const duplicate = (value, N) => {
   if (N <= 0) return [];
-  else {
-    res = [];
-    for (let i = 0; i < N; i++) {
-      res[i] = value;
-    }
-    return res;
-  }
+  return Array(N).fill(value);
```
Comment (tshemsedinov) [inline Tasks/duplicate.js:10]: Naming not ok

File: Tasks/except.js [hunk 6]
```diff
@@ -1,19 +1,20 @@
...
+  for (const key of keysToKeep) {
+    if (keysToRemove.includes(key)) {
+      delete incomingValuesArray[key];
     }
-  }, 2000);
-  ({ key: 'value' });
-  return incomingvaluesarray;
+  }
+
+  return incomingValuesArray;
```
Comment (tshemsedinov) [inline Tasks/except.js:17]: Do not change incoming args

File: Tasks/except.js [hunk 7]
```diff
@@ -1,19 +1,20 @@
...
+// replace forEach with for...of
+
+'use strict';
+
+const except = (incomingValuesArray, ...keysToRemove) => {
+  const keysToKeep = Object.keys(incomingValuesArray);
+
+  for (const key of keysToKeep) {
+    if (keysToRemove.includes(key)) {
+      delete incomingValuesArray[key];
```
Comment (tshemsedinov) [inline Tasks/except.js:13]: Try to avoid delete

File: Tasks/inc.js [hunk 8]
```diff
@@ -1,12 +1,18 @@
...
+
+const incrementNumbers = (dictionary) => {
+  for (const key in dictionary) {
+    if (typeof dictionary[key] === 'number') {
+      dictionary[key] = dictionary[key] + 1;
     }
   }
-  return format_complete;
+
+  return dictionary;
```
Comment (tshemsedinov) [inline Tasks/inc.js:15]: Do not change incoming args

File: Tasks/intersection.js [hunk 9]
```diff
@@ -1,25 +1,24 @@
...
+  const keysDictionary1 = Object.keys(dictionary1);
+
+  for (const key of keysDictionary1) {
+    if (dictionary1[key] === dictionary2[key]) {
+      dictionary2[key] = dictionary1[key];
     } else {
-      delete object_1[attribute_name];
-      {
-      }
+      delete dictionary1[key];
```
Comment (tshemsedinov) [inline Tasks/intersection.js:15]: ditto

File: Tasks/intersection.js [hunk 10]
```diff
@@ -1,25 +1,24 @@
...
-    {
-    }
-  }
-  {
   }
-  return object_1;
+
+  const intersection = dictionary1;
+  return intersection;
 };
```
Comment (tshemsedinov) [inline Tasks/intersection.js:21]: Looks strange

File: Tasks/ip.js [hunk 11]
```diff
@@ -1,19 +1,26 @@
...
+
+'use strict';
+
+const parseIP = (ipString) => {
+  const ipAddressArray = [];
+
+  if (!ipString) return;
+
+  const ipOctets = ipString.split('.');
+  if (ipOctets.length != 4) return;
```
Comment (tshemsedinov) [inline Tasks/ip.js:14]: Magic number

File: Tasks/between.js (hunk 1 above)
Comment (yuliakinda) [reply Tasks/between.js:13]: Fixed

File: Tasks/between.js (hunk 2 above)
Comment (yuliakinda) [reply Tasks/between.js:18]: Fixed

File: Tasks/between.js (hunk 3 above)
Comment (yuliakinda) [reply Tasks/between.js:23]: Fixed

File: Tasks/duplicate.js (hunk 5 above)
Comment (yuliakinda) [reply Tasks/duplicate.js:10]: Fixed

File: Tasks/except.js (hunk 6 above)
Comment (yuliakinda) [reply Tasks/except.js:17]: Fixed

File: Tasks/except.js (hunk 7 above)
Comment (yuliakinda) [reply Tasks/except.js:13]: Fixed

File: Tasks/intersection.js (hunk 9 above)
Comment (yuliakinda) [reply Tasks/intersection.js:15]: Fixed

File: Tasks/intersection.js (hunk 10 above)
Comment (yuliakinda) [reply Tasks/intersection.js:21]: Fixed

File: Tasks/ip.js (hunk 11 above)
Comment (yuliakinda) [reply Tasks/ip.js:14]: Fixed

**Pull request closed without merging**